spec/queries/admin/gdpr_delete_requests_query_spec.rb

```diff
@@ -0,0 +1,46 @@
+require "rails_helper"
+
+RSpec.describe Admin::GDPRDeleteRequestsQuery, type: :query do
+  subject { described_class.call(relation: ::GDPRDeleteRequest.all, search: nil) }
```

Comment:
Without the `search` variable (e.g. `search: search`) your queries below are never applying the `let(:search)` values you set.  You're essentially always calling `described_class.call(search: nil)`.  The `expect().to include` is passing because it doesn't check if it is only the given value(s), just if the expected value is within the enumerable.

What I've done in other cases is to "cheat" a bit.

```ruby
subject { described_class.call(search: search).to_a.sort }
```

This will produce an array of those GDPR requests that you can then compare as follows:

```ruby
it { is_expected.to match_array([a,b,c].sort) }
```

Also in the above example I removed the `relation: ::GDPRDeleteRequest.all` as it's redundent; you're already using that as the default parameter.

Reply:
So this is strange: the specs are consistently failing due to an actual collection and expected collection mismatch--the specs are receiving empty arrays when the arrays _should_ contain the GDPR Delete Requests that result in the search term given. It almost seems like something is not set up or loading properly in the spec, however, I cannot pinpoint what I am doing wrong. I have dropped into `pry` countless times to check what is being returned vs. what we expect in the tests, but I am still stumped. Any thoughts on what I might be overlooking?

cc @Ridhwana